lib/read_only/lfric/read_only.f90

```diff
@@ -0,0 +1,835 @@
+
+        !> The generic interface for providing the value of variables,
+        !! which in case of the NetCDF interface is written:                                               
+        generic, public :: ProvideVariable => ChecksumScalarInt,    &
+                                              ChecksumArray1dInt,   &
+                                              ChecksumArray2dInt,   &
+                                              ChecksumArray3dInt,   &
+                                              ChecksumScalarReal,   &
+                                              ChecksumArray1dDouble,&
+                                              ChecksumArray3dDouble,&
+                                              ChecksumArray4dDouble,&
+                                              ChecksumScalarDouble, &
+                                              ChecksumFieldDouble,  &
+                                              ChecksumFieldVectorDouble
+
+    end type read_only_verify_PSyDataType
+
+Contains
+
+    ! -------------------------------------------------------------------------
+    !> This subroutine is the first function called when an instrumented region
+    !! is entered. It initialises this object, and stores module and regin
+    !! names. 
+    !! @param[inout] this The instance of the read_only_verify_PSyDataType.
+    !! @param[in] module_name The name of the module of the instrumented
+    !!            region.
+    !! @param[in] kernel_name The name of the instrumented region.
+    !! @param[in] num_pre_vars The number of variables that are declared and
+    !!            written before the instrumented region.
+    !! @param[in] num_post_vars The number of variables that are also declared
+    !!            before an instrumented region of code, but are written after
+    !!            this region.
+    subroutine PreStart(this, module_name, region_name, num_pre_vars, &
+                        num_post_vars)
+        implicit none
+        class(read_only_verify_PSyDataType), intent(inout), target :: this
+        character(*), intent(in) :: module_name, region_name
+        integer, intent(in)      :: num_pre_vars, num_post_vars
+
+        character(1) :: verbose
+        integer :: status
+
+        call get_environment_variable("PSYDATA_VERBOSE", verbose, status=status)
+        this%verbosity=0
+        if (status==0) then
+            if(verbose=="0") then
+                this%verbosity = 0
+            else if (verbose=="1") then
+                this%verbosity = 1
+            else if (verbose=="2") then
+                this%verbosity = 2
+            else
+                print *,"PSyData: invalid setting of PSYDATA_VERBOSE."
+                print *,"It must be '0', 1' or '2', but it is '", verbose,"'."
+                this%verbosity = 0
+            endif
+        endif
+
+        if (num_pre_vars /= num_post_vars) then
+            print *,"PSYDATA: The same number of variables must be provided before"
```

Comment:
I checked, LFRic is using stop at this stage, I couldn't find anything else. There is a log call that will abort, but I am not sure if we want to add this dependency to log(?) - let me know.

For gocean I used the corresponding call.

Reply:
(to explain my issues with log: iirc it would be best if the psydata libraries only depend on infrastructure code - and imho I am not sure if log should be part of that).